gym/envs/registration.py

```diff
@@ -62,31 +70,33 @@ class EnvSpec:
 
     """
 
-    def __init__(
-        self,
-        id: str,
-        entry_point: Union[Callable, str, None] = None,
-        reward_threshold: Optional[int] = None,
-        nondeterministic: bool = False,
-        max_episode_steps: Optional[int] = None,
-        order_enforce: Optional[int] = True,
-        kwargs: dict = None,
-    ):
-        self.id = id
-        self.entry_point = entry_point
-        self.reward_threshold = reward_threshold
-        self.nondeterministic = nondeterministic
-        self.max_episode_steps = max_episode_steps
-        self.order_enforce = order_enforce
-        self._kwargs = {} if kwargs is None else kwargs
-
-        match = env_id_re.fullmatch(id)
-        if not match:
-            raise error.Error(
-                f"Attempted to register malformed environment ID: {id}."
-                f"(Currently all IDs must be of the form {env_id_re.pattern}.)"
-            )
-        self._env_name = match.group("name")
+    id_requested: InitVar[str]
+    entry_point: Optional[Union[Callable, str]] = field(default=None)
+    reward_threshold: Optional[int] = field(default=None)
+    nondeterministic: bool = field(default=False)
+    max_episode_steps: Optional[int] = field(default=False)
+    order_enforce: Optional[bool] = field(default=True)
```

Comment:
Is this optional? It was before this PR but I don't believe it is. Can you remove `Optional`.